### PR TITLE
[MM-25659] ability to permanent delete channel through client

### PR DESCRIPTION
**Pull request by ashishbhate**
#### Summary

- Ability to permanent delete channel through client
#### Ticket Link

- https://mattermost.atlassian.net/browse/MM-25659

### CODE REVIEW

File: api4/channel_local.go [hunk 1]
```diff
@@ -206,3 +206,40 @@ func localPatchChannel(c *Context, w http.ResponseWriter, r *http.Request) {
...
+	defer c.LogAuditRec(auditRec)
+	auditRec.AddMeta("channeld", channel)
+
+	if channel.Type == model.CHANNEL_DIRECT || channel.Type == model.CHANNEL_GROUP {
+		c.Err = model.NewAppError("localDeleteChannel", "api.channel.delete_channel.type.invalid", nil, "", http.StatusBadRequest)
+		return
+	}
+
+	if c.Params.Permanent {
+		err = c.App.PermanentDeleteChannel(channel)
```
Comment (isacikgoz) [inline api4/channel_local.go:232]: Why this is not respecting to `EnableAPIChannelDeletion`?

Comment (ashishbhate) [reply api4/channel_local.go:232]: @isacikgoz local mode is like super admin, so its not meant to respect this option.

Comment (isacikgoz) [reply api4/channel_local.go:232]: I see, I wonder if this should mentioned in docs or elsewhere. I would expect the local mode behavior is same with the authenticated system admin over tcp. Any ideas @thefactremains?

Comment (ashishbhate) [reply api4/channel_local.go:232]: I've added documentation PR for the similar `EnableAPITeamDeletion` option - https://github.com/mattermost/docs/pull/3767/

I'll open docs for this PR as well, after it is reviewed.

File: api4/channel.go [hunk 2]
```diff
@@ -1074,7 +1074,15 @@ func deleteChannel(c *Context, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	err = c.App.DeleteChannel(channel, c.App.Session().UserId)
+	if c.Params.Permanent {
```
Comment (mgdelacroix) [inline api4/channel.go:1077]: Is there a PR to document this flag in the API docs?

Comment (ashishbhate) [reply api4/channel.go:1077]: Yes, will open a few documentation PRs soon

File: api4/channel_local.go (hunk 1 above)
Comment (ashishbhate) [reply api4/channel_local.go:232]: I've opened PRs for docs:

https://github.com/mattermost/docs/pull/3825
https://github.com/mattermost/mattermost-api-reference/pull/563

File: api4/channel.go (hunk 2 above)
Comment (ashishbhate) [reply api4/channel.go:1077]: https://github.com/mattermost/mattermost-api-reference/pull/563

**Comment by amyblais** [comment]
@ashishbhate Does the new config setting need to be added to telemetry?

**Comment by ashishbhate** [comment]
@amyblais I've created a PR to add them to telemetry https://github.com/mattermost/mattermost-server/pull/15506

**Comment by justinegeffen** [comment]
I've checked the docs for this and can confirm that the following PRs have been merged:
- https://github.com/mattermost/docs/pull/3767
- https://github.com/mattermost/docs/pull/3825

This is awaiting merge: 
- https://github.com/mattermost/mattermost-api-reference/pull/563

The Telemetry PR is here: 
- https://github.com/mattermost/mattermost-server/pull/15506

**Pull request merged**